### ISSUE TITLE
NoClassDefFoundError: com.stripe.stripeterminal.storage.StripeTerminalDatabase after updating from beta 9 to rc-1

### ISSUE THREAD

**Issue by anton8900** [stripe/stripe-terminal-android#64]
09-17 11:55:45.863 10200-10200/au.com.ticketbooth.rfid.android E/com.stripe.stripeterminal.storage.StripeTerminalDatabase: error
    java.lang.NoClassDefFoundError: com.stripe.stripeterminal.storage.StripeTerminalDatabase
        at com.stripe.stripeterminal.Terminal$Companion.initTerminal(Terminal.kt:416)
        at com.stripe.stripeterminal.Terminal.initTerminal(Terminal.kt)

**Comment by chander-stripe** [comment]
Does this only happen on app upgrades or does it happen on a completely fresh install as well?

**Comment by chander-stripe** [comment]
Can I also see how you're calling `initTerminal` ?

**Comment by anton8900** [comment]
it's fail always doesn't matter app updated or intsalled after uninstall

**Comment by anton8900** [comment]
Terminal.initTerminal(application.getApplicationContext(), LogLevel.VERBOSE, new TokenProvider(), terminalEventListener);

**Comment by anton8900** [comment]
private static final TerminalEventListener terminalEventListener = new TerminalEventListener();

**Comment by anton8900** [comment]
package au.com.ticketbooth.rfid.android.service.stripe;

import com.google.common.collect.Lists;
import com.stripe.stripeterminal.callable.TerminalListener;
import com.stripe.stripeterminal.model.external.*;

import java.util.List;

import au.com.ticketbooth.rfid.android.mvp.presenter.settings.StripeReaderConnectPresenter;
import au.com.ticketbooth.rfid.android.ui.activities.settings.stripe.StripeReaderConnectActivity;

/**
 * The {@code TerminalEventListener} implements the {@link TerminalListener} interface and will
 * forward along any events to other parts of the app that register for updates.
 */
public class TerminalEventListener implements TerminalListener {

    private static List<TerminalListener> listeners = Lists.newArrayList();

    public static ConnectionStatus currentConnectionStatus = ConnectionStatus.NOT_CONNECTED;
    public static PaymentStatus currentPaymentStatus = PaymentStatus.NOT_READY;

    @Override
    public void onReportReaderEvent(ReaderEvent event) {
        for (TerminalListener listener : listeners) {
            listener.onReportReaderEvent(event);
        }
    }

    @Override
    public void onReportLowBatteryWarning() {
        for (TerminalListener listener : listeners) {
            listener.onReportLowBatteryWarning();
        }
    }

    @Override
    public void onUnexpectedReaderDisconnect(Reader reader) {
        StripeReaderConnectPresenter.READER_CONNECTED = false;
        for (TerminalListener listener : listeners) {
            listener.onUnexpectedReaderDisconnect(reader);
        }
    }

    @Override
    public void onConnectionStatusChange(ConnectionStatus status) {
        currentConnectionStatus = status;
        if(currentConnectionStatus != ConnectionStatus.CONNECTED) {
            StripeReaderConnectPresenter.READER_CONNECTED = false;
        }
        for (TerminalListener listener : listeners) {
            listener.onConnectionStatusChange(status);
        }
    }

    @Override
    public void onPaymentStatusChange(PaymentStatus status) {
        currentPaymentStatus = status;
        for (TerminalListener listener : listeners) {
            listener.onPaymentStatusChange(status);
        }
    }

    /**
     * @return the current ConnectionStatus
     */
    public static ConnectionStatus getCurrentConnectionStatus() {
        return currentConnectionStatus;
    }

    /**
     * @return the current PaymentStatus
     */
    public static PaymentStatus getCurrentPaymentStatus() {
        return currentPaymentStatus;
    }
}

**Comment by anton8900** [comment]
this worked perfect before updating to the rc-1

**Comment by chander-stripe** [comment]
Hmm, so the major new thing with Rc1 is that we're creating a Room database to store pending transactions and other traces to make sure that we handle any edge cases when the app is unexpectedly closed / re-opened - 

Does your app meet the standard requirements for a room database?

**Comment by legomanaustralia** [comment]
@chander-stripe so we tried test application on phone and its working so the phon is 100% compatible. There must be an incompatibility with our app.

What's the standard requirements for a room database here? I did a quick google search but couldn't find any reference to a definitive list of requirements. I'm guessing it could be due to out other ORM (dbflow) but tonnes of other applications use dbflow so it would seem strange that we'd strike all those from the table.
